tests/src/test/java/io/streamnative/pulsar/handlers/kop/DifferentNamespaceTestBase.java

```diff
@@ -99,8 +98,7 @@ void testSimpleProduceAndConsume(String topic) {
                     log.info("Successfully send {} to {}-partition-{}",
                             key, recordMetadata.topic(), recordMetadata.partition());
                 } else {
-                    log.error("Failed to send {}", key);
-                    fail("Failed to send " + key);
+                    log.error("Failed to send {}: {}", key, e.getMessage());
```

Comment:
why delete fail check?

Reply:
The fail check in callback can't stop the test immediately. It only makes the thread broken and prints unrelated logs. Here we send these messages asynchronously so we also don't want to stop the test if one produce failed. Otherwise here we should use synchronously send. (Whether to use async send here is another issue)

Removing the `fail` check also helps figuring out all messages' send result for debugging.

Reply:
No matter if `fail` check exists, if any send failed, the test will eventually fail for timeout because consumer cannot receive `numMessages` messages in test's timeout.

Reply:
we'd better use try... catch to deal with the produce exception instead of use callback mode.